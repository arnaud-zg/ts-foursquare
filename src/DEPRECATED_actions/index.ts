import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
