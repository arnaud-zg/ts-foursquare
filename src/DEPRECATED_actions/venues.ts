import { ActionType, createAsyncAction } from 'typesafe-actions'
import { NRequest } from '../../types'
import {
  adaptGetVenuesCategories,
  adaptGetVenuesExplore,
  adaptGetVenuesLikes,
  adaptGetVenuesListed,
  adaptGetVenuesNextVenues,
  adaptGetVenuesSearch,
  adaptGetVenuesSimilar,
  adaptGetVenuesSuggestCompletion,
  adaptGetVenuesTrending,
} from '../adapter/venues'

export const getVenuesCategoriesAsync = createAsyncAction(
  'GET_VENUES_CATEGORIES_REQUEST',
  'GET_VENUES_CATEGORIES_SUCCESS',
  'GET_VENUES_CATEGORIES_FAILURE',
  'GET_VENUES_CATEGORIES_CANCEL'
)<void, ReturnType<typeof adaptGetVenuesCategories>, Error, string>()

export const getVenuesExploreAsync = createAsyncAction(
  'GET_VENUES_EXPLORE_REQUEST',
  'GET_VENUES_EXPLORE_SUCCESS',
  'GET_VENUES_EXPLORE_FAILURE',
  'GET_VENUES_EXPLORE_CANCEL'
)<
  NRequest.TVenuesExplorePayload,
  ReturnType<typeof adaptGetVenuesExplore>,
  Error,
  string
>()

export const getVenuesLikesAsync = createAsyncAction(
  'GET_VENUES_LIKES_REQUEST',
  'GET_VENUES_LIKES_SUCCESS',
  'GET_VENUES_LIKES_FAILURE',
  'GET_VENUES_LIKES_CANCEL'
)<
  NRequest.IVenuesLikesPayload,
  ReturnType<typeof adaptGetVenuesLikes>,
  Error,
  string
>()

export const getVenuesListedAsync = createAsyncAction(
  'GET_VENUES_LISTED_REQUEST',
  'GET_VENUES_LISTED_SUCCESS',
  'GET_VENUES_LISTED_FAILURE',
  'GET_VENUES_LISTED_CANCEL'
)<
  NRequest.IVenuesListedPayload,
  ReturnType<typeof adaptGetVenuesListed>,
  Error,
  string
>()

export const getVenuesNextVenuesAsync = createAsyncAction(
  'GET_VENUES_NEXT_VENUES_REQUEST',
  'GET_VENUES_NEXT_VENUES_SUCCESS',
  'GET_VENUES_NEXT_VENUES_FAILURE',
  'GET_VENUES_NEXT_VENUES_CANCEL'
)<
  NRequest.IVenuesNextVenuesPayload,
  ReturnType<typeof adaptGetVenuesNextVenues>,
  Error,
  string
>()

export const getVenuesSearchAsync = createAsyncAction(
  'GET_VENUES_SEARCH_REQUEST',
  'GET_VENUES_SEARCH_SUCCESS',
  'GET_VENUES_SEARCH_FAILURE',
  'GET_VENUES_SEARCH_CANCEL'
)<
  NRequest.TVenuesSearchPayload,
  ReturnType<typeof adaptGetVenuesSearch>,
  Error,
  string
>()

export const getVenuesSimilarAsync = createAsyncAction(
  'GET_VENUES_SIMILAR_REQUEST',
  'GET_VENUES_SIMILAR_SUCCESS',
  'GET_VENUES_SIMILAR_FAILURE',
  'GET_VENUES_SIMILAR_CANCEL'
)<
  NRequest.IVenuesSimilarPayload,
  ReturnType<typeof adaptGetVenuesSimilar>,
  Error,
  string
>()

export const getVenuesSuggestCompletionAsync = createAsyncAction(
  'GET_VENUES_SUGGEST_COMPLETION_REQUEST',
  'GET_VENUES_SUGGEST_COMPLETION_SUCCESS',
  'GET_VENUES_SUGGEST_COMPLETION_FAILURE',
  'GET_VENUES_SUGGEST_COMPLETION_CANCEL'
)<
  NRequest.TVenuesSuggestCompletionPayload,
  ReturnType<typeof adaptGetVenuesSuggestCompletion>,
  Error,
  string
>()

export const getVenuesTrendingAsync = createAsyncAction(
  'GET_VENUES_TRENDING_REQUEST',
  'GET_VENUES_TRENDING_SUCCESS',
  'GET_VENUES_TRENDING_FAILURE',
  'GET_VENUES_TRENDING_CANCEL'
)<
  NRequest.TVenuesTrendingPayload,
  ReturnType<typeof adaptGetVenuesTrending>,
  Error,
  string
>()

export const venuesActions = {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
}

export type TVenuesAction = ActionType<typeof venuesActions>
