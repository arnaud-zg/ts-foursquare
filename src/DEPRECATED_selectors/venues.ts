import { NStore, NVenue } from '../../types'

export const venueCategoriesStateSelector = (
  state: NStore.IState,
  venueId: string
): NVenue.ICategory[] => {
  const venueEntity = venuesEntitiesSelector(state)[venueId]
  return venueEntity ? venueEntity.categories : []
}

export const venueLocationSelector = (
  state: NStore.IState,
  venueId: string
): NVenue.ILocation | null => {
  const venueEntity = venuesEntitiesSelector(state)[venueId]
  return venueEntity ? venueEntity.location : null
}

export const venuesCategoriesSelector = (state: NStore.IState) =>
  venuesSelector(state).categories

export const venueSelector = (state: NStore.IState, venueId: string) =>
  venuesEntitiesSelector(state)[venueId]

export const venuesEntitiesSelector = (state: NStore.IState) =>
  venuesSelector(state).entities

export const venuesNextVenuesSelector = (state: NStore.IState) =>
  venuesSelector(state).nextVenues

export const venuesSimilarSelector = (state: NStore.IState) =>
  venuesSelector(state).similarVenues

export const venuesRecommendedPlacesSelector = (state: NStore.IState) =>
  venuesSelector(state).recommendedPlaces

export const venuesListedSelector = (state: NStore.IState) =>
  venuesSelector(state).venuesListed

export const venuesListedGroupSelector = (
  state: NStore.IState,
  groupKey: string
) => venuesSelector(state).venuesListed[groupKey]

export const venuesSelector = (state: NStore.IState) => state.venues
