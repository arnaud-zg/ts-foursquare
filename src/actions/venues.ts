import { ActionType, createAsyncAction } from 'typesafe-actions'
import {
  NLikes,
  NMiniVenue,
  NRecommendedPlaces,
  NRequest,
  NVenue,
  NVenuesCategories,
} from '../../types'

export const getVenuesCategoriesAsync = createAsyncAction(
  'GET_VENUES_CATEGORIES_REQUEST',
  'GET_VENUES_CATEGORIES_SUCCESS',
  'GET_VENUES_CATEGORIES_FAILURE',
  'GET_VENUES_CATEGORIES_CANCEL'
)<void, NVenuesCategories.ICategory[], Error, string>()

export const getVenuesExploreAsync = createAsyncAction(
  'GET_VENUES_EXPLORE_REQUEST',
  'GET_VENUES_EXPLORE_SUCCESS',
  'GET_VENUES_EXPLORE_FAILURE',
  'GET_VENUES_EXPLORE_CANCEL'
)<
  NRequest.TVenuesExplorePayload,
  NRecommendedPlaces.IGroupItem[],
  Error,
  string
>()

export const getVenuesLikesAsync = createAsyncAction(
  'GET_VENUES_LIKES_REQUEST',
  'GET_VENUES_LIKES_SUCCESS',
  'GET_VENUES_LIKES_FAILURE',
  'GET_VENUES_LIKES_CANCEL'
)<NRequest.IVenuesLikesPayload, NLikes.ILikes, Error, string>()

export const getVenuesNextVenuesAsync = createAsyncAction(
  'GET_VENUES_NEXT_VENUES_REQUEST',
  'GET_VENUES_NEXT_VENUES_SUCCESS',
  'GET_VENUES_NEXT_VENUES_FAILURE',
  'GET_VENUES_NEXT_VENUES_CANCEL'
)<NRequest.IVenuesNextVenuesPayload, NVenue.IVenue[], Error, string>()

export const getVenuesSearchAsync = createAsyncAction(
  'GET_VENUES_SEARCH_REQUEST',
  'GET_VENUES_SEARCH_SUCCESS',
  'GET_VENUES_SEARCH_FAILURE',
  'GET_VENUES_SEARCH_CANCEL'
)<NRequest.TVenuesSearchPayload, NVenue.IVenue[], Error, string>()

export const getVenuesSimilarAsync = createAsyncAction(
  'GET_VENUES_SIMILAR_REQUEST',
  'GET_VENUES_SIMILAR_SUCCESS',
  'GET_VENUES_SIMILAR_FAILURE',
  'GET_VENUES_SIMILAR_CANCEL'
)<NRequest.IVenuesSimilarPayload, NVenue.IVenue[], Error, string>()

export const getVenuesTrendingAsync = createAsyncAction(
  'GET_VENUES_TRENDING_REQUEST',
  'GET_VENUES_TRENDING_SUCCESS',
  'GET_VENUES_TRENDING_FAILURE',
  'GET_VENUES_TRENDING_CANCEL'
)<NRequest.TVenuesTrendingPayload, NVenue.IVenue[], Error, string>()

export const getVenuesSuggestCompletionAsync = createAsyncAction(
  'GET_VENUES_SUGGEST_COMPLETION_REQUEST',
  'GET_VENUES_SUGGEST_COMPLETION_SUCCESS',
  'GET_VENUES_SUGGEST_COMPLETION_FAILURE',
  'GET_VENUES_SUGGEST_COMPLETION_CANCEL'
)<
  NRequest.TVenuesSuggestCompletionPayload,
  NMiniVenue.IMiniVenue[],
  Error,
  string
>()

export const venuesActions = {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
}

export type TVenuesAction = ActionType<typeof venuesActions>
