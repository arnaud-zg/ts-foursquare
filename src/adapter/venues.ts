import { NHttpStatuses } from 'http-response-status'
import { NPayload } from '../../types/payload.d'
import { NRecommendedPlaces } from '../../types/recommendedPlaces'
import { NVenue } from '../../types/venue.d'

export const adaptGetVenuesSearch = (
  payload: NPayload.IPayload<NVenue.IResponse>
): NVenue.IVenue[] =>
  !!payload.meta &&
  payload.meta.code === NHttpStatuses.ESuccess.OK &&
  !!payload.response &&
  !!payload.response.venues &&
  !!payload.response.venues.length
    ? payload.response.venues
    : []

export const adaptGetVenuesExplore = (
  payload: NPayload.IPayload<NRecommendedPlaces.IResponse>
): NRecommendedPlaces.IGroupItem[] =>
  !!payload.meta &&
  payload.meta.code === NHttpStatuses.ESuccess.OK &&
  !!payload.response &&
  !!payload.response.groups &&
  !!payload.response.groups.length &&
  !!payload.response.groups[0].items &&
  !!payload.response.groups[0].items.length
    ? payload.response.groups[0].items
    : []

export const adaptGetVenuesTrending = (
  payload: NPayload.IPayload<NVenue.IResponse>
): NVenue.IVenue[] => adaptGetVenuesSearch(payload)
