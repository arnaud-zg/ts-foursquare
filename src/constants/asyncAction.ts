import { getType } from 'typesafe-actions'
import {
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
} from '../DEPRECATED_actions'

interface IAsyncActionNameMapping {
  [actionType: string]: string
}

export const ASYNC_ACTION_NAME_MAPPING: IAsyncActionNameMapping = {
  // getVenuesCategoriesAsync
  [getType(getVenuesCategoriesAsync.request)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.success)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.failure)]: 'getVenuesCategoriesAsync',
  [getType(getVenuesCategoriesAsync.cancel)]: 'getVenuesCategoriesAsync',
  // getVenuesExploreAsync
  [getType(getVenuesExploreAsync.request)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.success)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.failure)]: 'getVenuesExploreAsync',
  [getType(getVenuesExploreAsync.cancel)]: 'getVenuesExploreAsync',
  // getVenuesLikesAsync
  [getType(getVenuesLikesAsync.request)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.success)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.failure)]: 'getVenuesLikesAsync',
  [getType(getVenuesLikesAsync.cancel)]: 'getVenuesLikesAsync',
  // getVenuesListedAsync
  [getType(getVenuesListedAsync.request)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.success)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.failure)]: 'getVenuesListedAsync',
  [getType(getVenuesListedAsync.cancel)]: 'getVenuesListedAsync',
  // getVenuesNextVenuesAsync
  [getType(getVenuesNextVenuesAsync.request)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.success)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.failure)]: 'getVenuesNextVenuesAsync',
  [getType(getVenuesNextVenuesAsync.cancel)]: 'getVenuesNextVenuesAsync',
  // getVenuesSearchAsync
  [getType(getVenuesSearchAsync.request)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.success)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.failure)]: 'getVenuesSearchAsync',
  [getType(getVenuesSearchAsync.cancel)]: 'getVenuesSearchAsync',
  // getVenuesSimilarAsync
  [getType(getVenuesSimilarAsync.request)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.success)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.failure)]: 'getVenuesSimilarAsync',
  [getType(getVenuesSimilarAsync.cancel)]: 'getVenuesSimilarAsync',
  // getVenuesSuggestCompletionAsync
  [getType(
    getVenuesSuggestCompletionAsync.request
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.success
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.failure
  )]: 'getVenuesSuggestCompletionAsync',
  [getType(
    getVenuesSuggestCompletionAsync.cancel
  )]: 'getVenuesSuggestCompletionAsync',
  // getVenuesTrendingAsync
  [getType(getVenuesTrendingAsync.request)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.success)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.failure)]: 'getVenuesTrendingAsync',
  [getType(getVenuesTrendingAsync.cancel)]: 'getVenuesTrendingAsync',
}
