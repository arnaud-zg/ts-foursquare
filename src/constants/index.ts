export const PROJECT_NAME = 'ts-foursquare'
