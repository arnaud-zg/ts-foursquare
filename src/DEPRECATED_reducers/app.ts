import { combineReducers } from 'redux'
import { initialState as initialStateLife, lifeReducer } from './life'
import { initialState as initialStateLists, listsReducer } from './lists'
import { initialState as initialStatePhotos, photosReducer } from './photos'
import { initialState as initialStateRequest, requestReducer } from './request'
import { initialState as initialStateStatus, statusReducer } from './status'
import { initialState as initialStateVenues, venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  lists: initialStateLists,
  photos: initialStatePhotos,
  request: initialStateRequest,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  lists: listsReducer,
  photos: photosReducer,
  request: requestReducer,
  status: statusReducer,
  venues: venuesReducer,
})
