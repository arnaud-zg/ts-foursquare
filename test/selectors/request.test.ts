import { getType } from 'typesafe-actions'
import { getVenuesSearchAsync } from '../../src/DEPRECATED_actions/venues'
import { ASYNC_ACTION_NAME_MAPPING } from '../../src/constants/asyncAction'
import { initialState } from '../../src/DEPRECATED_reducers/app'
import {
  requestActionSelector,
  requestCancelReasonSelector,
  requestEntityIdsSelector,
  requestErrorSelector,
  requestSelector,
} from '../../src/DEPRECATED_selectors/request'
import { NStore } from '../../types'
import { payload as payloadGetVenuesSearch } from '../epics/__mocks__/getVenuesSearchAsync.resolve'

const entityIds = payloadGetVenuesSearch.response.venues.map(item => item.id)

const state: NStore.IState = {
  ...initialState,
  request: {
    [ASYNC_ACTION_NAME_MAPPING[getType(getVenuesSearchAsync.success)]]: {
      entityIds,
    },
  },
}

describe('selectors/venues', () => {
  it('should get request state', () => {
    expect(requestSelector(state)).toMatchSnapshot()
  })

  it('should get request action', () => {
    expect(
      requestActionSelector(state, getType(getVenuesSearchAsync.success))
    ).toMatchSnapshot()
  })

  it("should get request action's entityIds", () => {
    expect(
      requestEntityIdsSelector(
        { ...initialState },
        getType(getVenuesSearchAsync.success)
      )
    ).toMatchSnapshot()
    expect(
      requestEntityIdsSelector(state, getType(getVenuesSearchAsync.success))
    ).toMatchSnapshot()
  })

  it("should get request action's error", () => {
    expect(
      requestErrorSelector(
        { ...initialState },
        getType(getVenuesSearchAsync.failure)
      )
    ).toMatchSnapshot()
    expect(
      requestErrorSelector(
        {
          ...state,
          request: {
            [ASYNC_ACTION_NAME_MAPPING[
              getType(getVenuesSearchAsync.success)
            ]]: {
              entityIds,
              error: new Error('An error occurred'),
            },
          },
        },
        getType(getVenuesSearchAsync.failure)
      )
    ).toMatchSnapshot()
  })

  it("should get request action's cancel reason", () => {
    expect(
      requestCancelReasonSelector(
        { ...initialState },
        getType(getVenuesSearchAsync.cancel)
      )
    ).toMatchSnapshot()
    expect(
      requestCancelReasonSelector(
        {
          ...state,
          request: {
            [ASYNC_ACTION_NAME_MAPPING[
              getType(getVenuesSearchAsync.success)
            ]]: {
              entityIds,
              cancelReason: 'An user cancelled this action',
            },
          },
        },
        getType(getVenuesSearchAsync.cancel)
      )
    ).toMatchSnapshot()
  })
})
