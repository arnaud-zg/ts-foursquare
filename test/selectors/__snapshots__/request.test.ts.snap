// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors/venues should get request action 1`] = `
Object {
  "entityIds": Array [
    "3fd66200f964a5209beb1ee3",
  ],
}
`;

exports[`selectors/venues should get request action's cancel reason 1`] = `undefined`;

exports[`selectors/venues should get request action's cancel reason 2`] = `
Object {
  "cancelReason": "An user cancelled this action",
  "entityIds": Array [
    "3fd66200f964a5209beb1ee3",
  ],
}
`;

exports[`selectors/venues should get request action's entityIds 1`] = `Array []`;

exports[`selectors/venues should get request action's entityIds 2`] = `
Object {
  "entityIds": Array [
    "3fd66200f964a5209beb1ee3",
  ],
}
`;

exports[`selectors/venues should get request action's error 1`] = `undefined`;

exports[`selectors/venues should get request action's error 2`] = `
Object {
  "entityIds": Array [
    "3fd66200f964a5209beb1ee3",
  ],
  "error": [Error: An error occurred],
}
`;

exports[`selectors/venues should get request state 1`] = `
Object {
  "getVenuesSearchAsync": Object {
    "entityIds": Array [
      "3fd66200f964a5209beb1ee3",
    ],
  },
}
`;
